File: main/search-for-a-range/search-for-a-range.cpp
class Solution {
public:
    static vector<int> searchRange(const vector<int>& nums, int target);
};

vector<int> Solution::searchRange(const vector<int>& nums, const int target)
{
    const auto beg = cbegin(nums);

    vector<int>::const_iterator left, right;
    tie(left, right) = equal_range(beg, cend(nums), target);

    if (left == right) return {-1, -1};
    return {left - beg, right - 1 - beg};
}

File: main/search-for-a-range/search-for-a-range.py
class Solution:
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        left = bisect.bisect_left(nums, target)
        right = bisect.bisect_right(nums, target, left)
        return (-1, -1) if left == right else (left, right - 1)
